src/adapter/src/coord/message_handler.rs

```diff
@@ -67,7 +67,44 @@ impl<S: Append + 'static> Coordinator<S> {
             Message::LinearizeReads(pending_read_txns) => {
                 self.message_linearize_reads(pending_read_txns).await;
             }
+            Message::StorageUsage => {
+                self.storage_usage_update().await;
+            }
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn storage_usage_update(&mut self) {
+        let object_id = None;
+        let shard_sizes = self.storageusageclient.shard_sizes().await;
+
+        let mut unk_storage = 0u64;
+        let mut known_storage = 0u64;
+        for (key, val) in shard_sizes.iter() {
+            if key.is_some() {
+                known_storage += val;
+                continue;
+            }
+            unk_storage += val
         }
+        // TODO: What, if anything, do we want to do with orphaned storage?
```

Comment:
I'm not entirely sure, but I think it's worth flagging, maybe something we alert on eventually? I would hope this is never above 0, but since storing data in S3 isn't free, it seems worth knowing if we're somehow paying for data that we're either miscategorizing or maybe it shouldn't exist